api/transport/inbound.go

```diff
@@ -25,31 +25,17 @@ package transport
 // Inbound is a transport that knows how to receive requests for procedure
 // calls.
 type Inbound interface {
+	Lifecycle
+
 	// SetRegistry configures the inbound to dispatch requests through a
 	// registry, typically called by a Dispatcher with its Registrar of handled
 	// procedures.
+	// Must be called before Inbound.Start() (or the inbound must already have
+	// a registry set
```

Comment:
Also, instead of this, do you want to say "Start MAY panic if SetRegistry was not called."

This lets inbound implementations decide if their Start wants to panic or if they have an implicit registry of some kind.